addons/sale/models/product_template.py

```diff
@@ -97,11 +97,11 @@ def _compute_service_tracking(self):
         super()._compute_service_tracking()
         self.filtered(lambda pt: not pt.sale_ok).service_tracking = 'no'
 
-    @api.depends('name')
+    @api.depends('name', 'purchase_ok')
     def _compute_visible_expense_policy(self):
         visibility = self.env.user.has_group('analytic.group_analytic_accounting')
         for product_template in self:
-            product_template.visible_expense_policy = visibility
+            product_template.visible_expense_policy = visibility and product_template.purchase_ok
```

Comment:
Seems a strange change to me :thinking:
Only purchasable products can be expensed ?

cc: @dalaOdoo

Reply:
In the spec we want it to be visible if 'expenses' or 'purchase' are enabled on the product

Reply:
Okay, I suppose I missed an override then :D